### PR TITLE
test code coverage to 100

### LINKED ISSUE #43: Coverage 100%
Improve corevage to meet 100%

### PR DESCRIPTION
**Pull request by dancastillo**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

closes #43

### CODE REVIEW

**Comment by Fdawgs** [comment]
Thanks @dancastillo, I think it'd also be worth updating `.taprc` to check and enforce the 100% coverage

**Review by mcollina — APPROVED** [review]
lgtm with @Fdawgs suggestion

**Comment by dancastillo** [comment]
@Fdawgs sounds good, `.taprc` should be updated now

**Review by Fdawgs — APPROVED** [review]

File: test/index.test.js [hunk 1]
```diff
@@ -710,3 +710,113 @@ test('onTimeout can be an async function', t => {
...
+
+  fastify.inject('/?error=true', (err, res) => {
+    t.error(err)
+  })
+
+  setTimeout(() => {
+    fastify.inject('/?error=false', (err, res) => {
+      t.equal(null, err)
+      t.equal(res.statusCode, 500)
+      t.same({ err: 'custom error' }, JSON.parse(res.payload))
```
Comment (Eomm) [inline test/index.test.js:755]:
```suggestion
      t.same(res.json(), { err: 'custom error' })
```

File: test/index.test.js [hunk 2]
```diff
@@ -710,3 +710,113 @@ test('onTimeout can be an async function', t => {
...
+      statusCode: 500
+    }, JSON.parse(res.payload))
+  })
+
+  fastify.inject('/?error=true', (err, res) => {
+    t.error(err)
+    t.equal(res.statusCode, 500)
+    t.same({
+      err: 'custom error'
+    }, JSON.parse(res.payload))
```
Comment (Eomm) [inline test/index.test.js:800]:
```suggestion
    t.same(res.json(), { err: 'custom error' })
```

File: test/index.test.js [hunk 3]
```diff
@@ -710,3 +710,113 @@ test('onTimeout can be an async function', t => {
...
+        statusCode: 500
+      }, JSON.parse(res.payload))
+    })
+
+    fastify.inject('/?error=true', (err, res) => {
+      t.equal(null, err)
+      t.equal(res.statusCode, 500)
+      t.same({
+        err: 'custom error'
+      }, JSON.parse(res.payload))
```
Comment (Eomm) [inline test/index.test.js:819]:
```suggestion
      t.same(res.json(), { err: 'custom error' })
```

**Review by Eomm — COMMENTED** [review]
nit the `wanted` object should be the 2nd parameter when we run a `t.same` check or the error messages would be confusing

**Pull request merged**